plugin/trino-opensearch/src/main/java/io/trino/plugin/opensearch/client/OpenSearchClient.java

```diff
@@ -368,7 +368,8 @@ public boolean indexExists(String index)
             return response.getStatusLine().getStatusCode() == 200;
         }
         catch (ResponseException e) {
-            if (e.getResponse().getStatusLine().getStatusCode() == 404) {
+            int statusCode = e.getResponse().getStatusLine().getStatusCode();
+            if (statusCode == 403 || statusCode == 404) {
```

Comment:
404 condition might be required to be added at below place as well.
https://github.com/trinodb/trino/blob/f3c71cade7e16331429de40cf051ac20979ac16b/plugin/trino-opensearch/src/main/java/io/trino/plugin/opensearch/OpenSearchMetadata.java#L455 as well

Reply:
This is called on the listTable method where we have already fixed the 403 issue. I can still add it here as well. Please confirm.